src/microbe_stage/MicrobeAI.cs

```diff
@@ -119,281 +74,109 @@ public void Think(float delta, Random random, MicrobeAICommonData data)
-           If it was ran in evaluate environment, it would only work if the microbe was in the neutral state.
-           because we may need more of these very specific things in the future for things like latching onto rocks */
-        // If you are predating and not being engulfed, don't run away until you switch state (keeps predators chasing
-        // you even when their predators are nearby) Its not a good survival strategy but it makes the game more fun.
-        if (predator != null && (lifeState != LifeState.PREDATING_STATE || microbe.IsBeingEngulfed))
+        // Don't bother with chunks when there's a lot of microbes to compete with
+        if (chosenChunk != null)
         {
-            try
+            var numRivals = 0;
+            foreach (var rival in allMicrobes)
             {
-                if (!predator.Dead)
+                if (rival != microbe)
                 {
-                    if ((microbe.Translation - predator.Translation).LengthSquared() <=
-                        (2000 + ((predator.HexCount * 8.0f) * 2)))
+                    var rivalDistance = (rival.Translation - chosenChunk.Translation).LengthSquared();
+                    if (rivalDistance < 500.0f &&
+                        rivalDistance < (microbe.Translation - chosenChunk.Translation).LengthSquared())
                     {
-                        if (lifeState != LifeState.FLEEING_STATE)
-                        {
-                            // Reset target position for faster fleeing
-                            hasTargetPosition = false;
-                        }
-
-                        boredom = 0;
-                        lifeState = LifeState.FLEEING_STATE;
+                        numRivals++;
                     }
                 }
-                else
-                {
-                    predator = null;
-                }
             }
-            catch (ObjectDisposedException)
-            {
-                // Our predator might be already disposed
-                predator = null;
-            }
-        }
-    }
-
-    /// <summary>
-    /// Gets the nearest chunk. And builds a list on chunkList
-    /// </summary>
-    /// <returns>The nearest chunk item.</returns>
-    /// <param name="allChunks">All chunks the AI knows of.</param>
-    private FloatingChunk GetNearestChunkItem(List<FloatingChunk> allChunks)
-    {
-        FloatingChunk chosenChunk = null;
-
-        Vector3 testPosition = new Vector3(0, 0, 0);
-        bool setPosition = true;
 
-        // Retrieve nearest potential chunk
-        foreach (var chunk in allChunks)
-        {
-            if (chunk == previousChunk)
-                continue;
-
-            if ((SpeciesOpportunism == Constants.MAX_SPECIES_OPPORTUNISM) ||
-                ((microbe.EngulfSize * (SpeciesOpportunism / Constants.OPPORTUNISM_DIVISOR)) >
-                    chunk.Size))
+            if (numRivals > 3)
```

Comment:
I think some randomness like between 1 to 5 (based on Random and the species opportunisim value?) would make it less likely that no cells try to grab some chunk

Reply:
I've added a calculation based on opportunism, but kept the random roll out. This method is called every thinking step, so rolling dice for each microbe might cause performance issues.

Reply:
Generating random numbers is much cheaper than calculating the nearby cells as there can be hundreds of alive cells at once. If anything there needs to be more target caching etc. so that all chunks and microbe distances aren't constantly recalculated.

Reply:
Added a touch of random, as requested: https://github.com/adQuid/Thrive/blob/improve_microbe_ai/src/microbe_stage/MicrobeAI.cs#L177